fluss-lake/fluss-lake-iceberg/src/test/java/com/apache/fluss/lake/iceberg/catalog/IcebergLakeCatalogTest.java

```diff
@@ -0,0 +1,334 @@
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.inmemory.InMemoryCatalog;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.Types;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.alibaba.fluss.metadata.TableDescriptor.BUCKET_COLUMN_NAME;
+import static com.alibaba.fluss.metadata.TableDescriptor.OFFSET_COLUMN_NAME;
+import static com.alibaba.fluss.metadata.TableDescriptor.TIMESTAMP_COLUMN_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+/** Unit test for {@link IcebergLakeCatalog}. */
+class IcebergLakeCatalogTest {
+
+    @TempDir private File tempWarehouseDir;
+
+    private Configuration configuration;
+    private Catalog icebergCatalog;
+    private SupportsNamespaces namespaceSupport;
+    private IcebergLakeCatalog flussIcebergCatalog;
+
+    @BeforeEach
+    void setupCatalog() {
+        configuration = new Configuration();
+        configuration.setString("warehouse", tempWarehouseDir.toURI().toString());
+
+        InMemoryCatalog inMemoryCatalog = new InMemoryCatalog();
+        inMemoryCatalog.initialize(
+                "test_catalog", ImmutableMap.of("warehouse", tempWarehouseDir.toURI().toString()));
+
+        this.icebergCatalog = inMemoryCatalog;
+        this.namespaceSupport = inMemoryCatalog;
+        this.flussIcebergCatalog = new IcebergLakeCatalog(inMemoryCatalog);
+    }
+
+    @Test
+    void verifiesBasicTableIdentifierAndWarehouseConfig() {
+        TableIdentifier tableId = TableIdentifier.of("test_db", "simple_table");
+
+        assertThat(tableId.name()).isEqualTo("simple_table");
+        assertThat(tableId.namespace().toString()).isEqualTo("test_db");
+
+        assertThat(configuration.toMap().get("warehouse"))
+                .isEqualTo(tempWarehouseDir.toURI().toString());
+        assertThat(icebergCatalog).isNotNull();
+    }
+
+    @Test
+    void createsTableInNewNamespaceSuccessfully() {
+        String database = "test_db";
+        String tableName = "simple_table";
+
+        createNamespaceIfAbsent(database);
+
+        TableIdentifier tableId = TableIdentifier.of(database, tableName);
+        org.apache.iceberg.Schema schema =
+                new org.apache.iceberg.Schema(
+                        Types.NestedField.required(1, "id", Types.IntegerType.get()),
+                        Types.NestedField.optional(2, "name", Types.StringType.get()));
+
+        Table createdTable = icebergCatalog.createTable(tableId, schema);
```

Comment:
We can remove this method. Actaully, we don't care about iceberg's catalog can create table or not since it's iceberg implementation. We need care about our `IcebergLakeCatalog` should create table successfully.